shark_turbine/kernel/wave/wave.py

```diff
@@ -118,6 +137,25 @@ def is_reduction(node: fx.Node):
                 if tiling_constraint.dim == custom.axis:
                     tiling_constraint.induction_var = self.induction_vars[custom]
 
+    def initialize_wave_constraints(self, trace: CapturedTrace) -> None:
+        """
+        For each wave constraint, determines the appropriate wave id by looking
+        for workgroup constraints along the same dimension and using information
+        from the hardware constraints.
```

Comment:
>    Because we represent the number of threads per block as
    [wave_id_0 * threads_per_wave, wave_id_1, wave_id_2], special care is
    required when computing wave_id_0. Specifically,
    wave_id_0 = floor(thread_id_0 / threads_per_wave)
    wave_id_1 = thread_id_1
    wave_id_2 = thread_id_2

Maybe should move this comment here

Reply:
Sure I can move it here.